go-controller/pkg/util/pod_annotation.go

```diff
@@ -261,24 +261,43 @@ func GetPodCIDRsWithFullMask(pod *v1.Pod) ([]*net.IPNet, error) {
 // GetAllPodIPs returns the pod's IP addresses, first from the OVN annotation
 // and then falling back to the Pod Status IPs. This function is intended to
 // also return IPs for HostNetwork and other non-OVN-IPAM-ed pods.
-func GetAllPodIPs(pod *v1.Pod) ([]net.IP, error) {
-	annotation, err := UnmarshalPodAnnotation(pod.Annotations, types.DefaultNetworkName)
-	if err == nil && annotation != nil {
+func GetAllPodIPs(pod *v1.Pod, nInfo NetInfo) ([]net.IP, error) {
+	// if netInfo is nil, this is called fro default network
+	ips := []net.IP{}
+	nadName := types.DefaultNetworkName
+	if nInfo.IsSecondary() {
+		on, network, err := IsNetworkOnPod(pod, nInfo)
+		if err != nil {
+			return nil, err
+		} else if !on {
+			// the pod is not attached to this specific network, don't return error
+			return ips, nil
+		}
+		nadName = GetNadName(network.Namespace, network.Name)
+	}
+	annotation, _ := UnmarshalPodAnnotation(pod.Annotations, nadName)
+	if annotation != nil {
 		// Use the OVN annotation if valid
-		ips := make([]net.IP, 0, len(annotation.IPs))
 		for _, ip := range annotation.IPs {
 			ips = append(ips, ip.IP)
 		}
 		// An OVN annotation should never have empty IPs, but just in case
-		if len(ips) > 0 {
-			return ips, nil
+		if len(ips) == 0 {
+			klog.Warningf("No IPs found in existing OVN annotation for nad %s! Pod Name: %s, Annotation: %#v",
+				nadName, pod.Name, annotation)
```

Comment:
if we move it down below, we'd need to check "if annotation != nil" instead, right?

Reply:
Hmm I thought that this warning would only make sense for the default network. Because for secondary networks we are already returning an error below that will be logged eventually as well. I will leave it up to you.